src/arkode/arkode_arkstep.c

```diff
@@ -2088,14 +2127,31 @@ int arkStep_TakeStep_Z(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr)
   if (*nflagPtr < 0) { return (*nflagPtr); }
   if (*nflagPtr > 0) { return (TRY_AGAIN); }
 
+  if (ark_mem->checkpoint_scheme && (*nflagPtr) == 0)
+  {
+    sunbooleantype do_save;
+    SUNAdjointCheckpointScheme_ShouldWeSave(ark_mem->checkpoint_scheme,
```

Comment:
I am not sure I follow. Are you suggesting that the InsertVector function should determine if the vector should actually be stored or discarded?

Reply:
> Are you suggesting that the InsertVector function should determine if the vector should actually be stored or discarded?

Yes. I don't see why this would prevent preparation for inserting a vector since it just combines two functions into one. Regardless, I don't feel strongly about this, so two functions is fine.

Reply:
>. I don't see why this would prevent preparation for inserting a vector since it just combines two functions into one.
You would need to first know if this is a step/stage to save before doing any preparation (otherwise you may do unnecessary work).